crc/services/workflow_processor.py

```diff
@@ -294,22 +268,22 @@ def get_parser():
         return parser
 
     @staticmethod
-    def get_spec(file_data_models: List[FileDataModel], workflow_spec_id):
+    def get_spec(files, workflow_spec_id):
         """Returns a SpiffWorkflow specification for the given workflow spec,
         using the files provided.  The Workflow_spec_id is only used to generate
         better error messages."""
         parser = WorkflowProcessor.get_parser()
         process_id = None
 
-        for file_data in file_data_models:
-            if file_data.file_model.type == FileType.bpmn:
-                bpmn: etree.Element = etree.fromstring(file_data.data)
-                if file_data.file_model.primary and file_data.file_model.workflow_spec_id == workflow_spec_id:
-                    process_id = FileService.get_process_id(bpmn)
-                parser.add_bpmn_xml(bpmn, filename=file_data.file_model.name)
-            elif file_data.file_model.type == FileType.dmn:
-                dmn: etree.Element = etree.fromstring(file_data.data)
-                parser.add_dmn_xml(dmn, filename=file_data.file_model.name)
+        for file in files:
+            if file['meta']['name'][-4:] == FileType.bpmn.value:
```

Comment:
hard coding the data structure here doesn't feel very clean.  would be good if we could get get the extension rather than trying to calcuate it.  And would be good to pass models around rather than json data structures.